### ISSUE TITLE
React Compatibility 

### ISSUE THREAD

**Issue by wudstrand** [TypeFox/monaco-languageclient#412]
I am able to get the client example to connect with my python LSP server, but when I try to use the library in my react app it isn't working. I was wondering if we could get some docs on how to integrate with react, or if anyone knew of any guides on how to connect a custom LSP server to the `react-monaco-editor`.

**Comment by kaisalmen** [comment]
Hi @wudstrand 

> I try to use the library in my react app it isn't working

Can you be more specific about that? What does not work? We don't have a react specific example, only the webpack related one to show how to use it.

**Comment by CGNonofr** [comment]
Are you using @monaco-editor/react or react-monaco-editor? `@monaco-editor/react` won't work as it doesn't use the ESM version

What is the result of a `npm list monaco-editor`?

**Comment by wudstrand** [comment]
I am using the `@monaco-editor/react` at the moment so I guess I need to transition over to `react-monaco-editor` library. Do I need to use web pack with this component? Here is a copy of my current impl of the editor in react. If you could let me know if I am on the right track that would be great.

```typescript
import React, {useState} from 'react';
import Editor, {EditorProps, Monaco} from "@monaco-editor/react";
import {toSocket, WebSocketMessageReader, WebSocketMessageWriter} from "vscode-ws-jsonrpc";
import {
    CloseAction,
    ErrorAction,
    MessageTransports,
    MonacoLanguageClient,
} from "monaco-languageclient";
import normalizeUrl from 'normalize-url';



const ReconnectingWebSocket = require('reconnecting-websocket');


interface Props extends EditorProps {
    enableEdit?: boolean
}

const LSPConnectedEditor: React.FC<Props> = ({enableEdit, ...other}) => {
    const [editor, setEditor] = useState<any | null>(null)

    function createLanguageClient (transports: MessageTransports): MonacoLanguageClient {
        return new MonacoLanguageClient({
            name: 'Sample Language Client',
            clientOptions: {
                // use a language id as a document selector
                documentSelector: ['python'],
                // disable the default error handler
                errorHandler: {
                    error: () => ({ action: ErrorAction.Continue }),
                    closed: () => ({ action: CloseAction.DoNotRestart })
                }
            },
            // create a language client connection from the JSON RPC connection on demand
            connectionProvider: {
                get: () => {
                    return Promise.resolve(transports);
                }
            }
        });
    }

    function createWebsocket(socketUrl: string) {
        const socketOptions = {
            maxReconnectionDelay: 10000,
            minReconnectionDelay: 1000,
            reconnectionDelayGrowFactor: 1.3,
            connectionTimeout: 10000,
            maxRetries: Infinity,
            debug: false
        };
        return new ReconnectingWebSocket.default(socketUrl, [], socketOptions);
    }

    function createUrl (hostname: string, port: number, path: string): string {
        // eslint-disable-next-line no-restricted-globals
        const protocol = location.protocol === 'https:' ? 'wss' : 'ws';
        return normalizeUrl(`${protocol}://${hostname}:${port}${path}`);
    }

    const beforeMount = (monaco: Monaco) => {
        monaco.languages.register({
            id: 'python',
            extensions: ['.py'],
            aliases: ['PYTHON', 'python', 'py'],
        });
    }

    const onMount = () => {

        // create the web socket
        const url = createUrl('localhost', 5000, '/');
        const webSocket = createWebsocket(url);

        webSocket.onopen = () => {
            const socket = toSocket(webSocket);
            const reader = new WebSocketMessageReader(socket);
            const writer = new WebSocketMessageWriter(socket);
            const languageClient = createLanguageClient({
                reader,
                writer
            });
            languageClient.start();
            reader.onClose(() => languageClient.stop());
        };
    }

    return (
        <Editor
            height="90vh"
            defaultLanguage={"python"}
            language={'python'}
            theme={'custom-theme'}
            beforeMount={(monaco: Monaco) => beforeMount(monaco)}
            onMount={(editor, monaco: Monaco) => onMount()}
            {...other}
        />
    )
}

export default LSPConnectedEditor;

```

Here is the output of `npm list monaco-editor`
```text
+-- @monaco-editor/react@4.4.5
| +-- @monaco-editor/loader@1.3.2
| | `-- monaco-editor@0.34.0 deduped
| `-- monaco-editor@0.34.0
+-- monaco-editor-webpack-plugin@7.0.1
| `-- monaco-editor@0.34.0 deduped
+-- monaco-editor-workers@0.34.1
| `-- monaco-editor@0.34.0 deduped
`-- monaco-languageclient@3.0.0
  `-- vscode@npm:@codingame/monaco-vscode-api@1.69.9       
    `-- monaco-editor@0.34.0 deduped
```

**Comment by CGNonofr** [comment]
Webpack is not required, you can use whatever you want as soon as everything use the ESM version of monaco-editor from npm.

**Comment by wudstrand** [comment]
Here is what I have so far. I am seeing an `Uncaught Error: Missing service editorService` error now.

![Capture](https://user-images.githubusercontent.com/10553702/186466045-d3b54f11-4216-4d0c-ac66-54c0e4e6bd74.PNG)

```typescript
import React, {useState} from 'react';
import {toSocket, WebSocketMessageReader, WebSocketMessageWriter} from "vscode-ws-jsonrpc";
import {
    CloseAction,
    ErrorAction,
    MessageTransports,
    MonacoLanguageClient,
} from "monaco-languageclient";
import normalizeUrl from 'normalize-url';
import MonacoEditor from "react-monaco-editor";
import {MonacoEditorProps} from "react-monaco-editor/src/types";
import * as monaco from "monaco-editor";


type Monaco = typeof monaco;
const ReconnectingWebSocket = require('reconnecting-websocket');


interface Props extends MonacoEditorProps {
    enableEdit?: boolean
}

const LSPConnectedEditor: React.FC<Props> = ({enableEdit, ...other}) => {
    const [editor, setEditor] = useState<any | null>(null)

    function createLanguageClient (transports: MessageTransports): MonacoLanguageClient {
        return new MonacoLanguageClient({
            name: 'Sample Language Client',
            clientOptions: {
                // use a language id as a document selector
                documentSelector: ['python'],
                // disable the default error handler
                errorHandler: {
                    error: () => ({ action: ErrorAction.Continue }),
                    closed: () => ({ action: CloseAction.DoNotRestart })
                }
            },
            // create a language client connection from the JSON RPC connection on demand
            connectionProvider: {
                get: () => {
                    return Promise.resolve(transports);
                }
            }
        });
    }

    function createWebsocket(socketUrl: string) {
        const socketOptions = {
            maxReconnectionDelay: 10000,
            minReconnectionDelay: 1000,
            reconnectionDelayGrowFactor: 1.3,
            connectionTimeout: 10000,
            maxRetries: Infinity,
            debug: false
        };
        return new ReconnectingWebSocket.default(socketUrl, [], socketOptions);
    }

    function createUrl (hostname: string, port: number, path: string): string {
        // eslint-disable-next-line no-restricted-globals
        const protocol = location.protocol === 'https:' ? 'wss' : 'ws';
        return normalizeUrl(`${protocol}://${hostname}:${port}${path}`);
    }

    const beforeMount = (monaco: Monaco) => {
        monaco.languages.register({
            id: 'python',
            extensions: ['.py'],
            aliases: ['PYTHON', 'python', 'py'],
        });
    }

    const onMount = () => {
        console.log('Mounted')

        // create the web socket
        const url = createUrl('localhost', 5000, '/');
        const webSocket = createWebsocket(url);

        webSocket.onopen = () => {
            console.log("Opening Web socket connection")
            const socket = toSocket(webSocket);
            const reader = new WebSocketMessageReader(socket);
            const writer = new WebSocketMessageWriter(socket);
            const languageClient = createLanguageClient({
                reader,
                writer
            });

            languageClient.start()
                .then(() => console.log('language client started'))
                .finally(() => console.log('language client done'));

            reader.onClose(() => languageClient.stop());
        };
    }

    return (
        <MonacoEditor
            height="90vh"
            language={'python'}
            theme={'custom-theme'}
            editorWillMount={(monaco: Monaco) => beforeMount(monaco)}
            editorDidMount={(editor, monaco: Monaco) => onMount()}
            {...other}
        />
    )
}

export default LSPConnectedEditor;

```

**Comment by CGNonofr** [comment]
The issue is that `react-monaco-editor` has `monaco-editor` v0.33 as dependency while monaco-languageclient v3 REQUIRES monaco-editor 0.34, try to force the update to monaco-editor 0.34

**Comment by kaisalmen** [comment]
@CGNonofr we should add a table to the main README stating the version requirements. I was thinking about that before and will add something soon.

I was also thinking whether makes sense to have usage examples for react, angular, vue and webpack outside this repo. First, we can move unneeded things out of here (removing unneeded dependencies from the examples) and to be able to point people to existing resources. I will see if I do this in the upcoming weeks whenever there is time (lower prio).

**Comment by CGNonofr** [comment]
> we should add a table to the main README stating the version requirements

It can't hurt :+1:

> I was also thinking whether makes sense to have usage examples for react, angular, vue and webpack outside this repo

Why does it need to be outside of the repo?

**Comment by kaisalmen** [comment]
> Why does it need to be outside of the repo?

Yeah, god question. Maybe it is better to create two workspaces: one for the two libs and another on for the examples. Lean dependencies for the libs and whatever is needed can go in the examples.

**Comment by CGNonofr** [comment]
> Yeah, god question

![Cima_da_Conegliano,_God_the_Father](https://user-images.githubusercontent.com/23257479/186628553-5642753b-2b19-4e38-b1c5-5e525419837f.jpg)

**Comment by kaisalmen** [comment]
🤣 Good, of course

**Comment by CGNonofr** [comment]
I'm not sure why the angular/react demos can't be along with the other demos though

**Comment by kaisalmen** [comment]
> I'm not sure why the angular/react demos can't be along with the other demos though

Yes, they can. I would just like to have separated npm workspaces, one for `monaco-languageclient` and `vscode-ws-jsonrpc` and one for all current and future examples.

**Comment by kaisalmen** [comment]
I know that I threw it all together a couple, but I now think it is better to have a separation here.

**Comment by colinblaise** [comment]
My issue may be different, but I also started receiving the following error:

![image](https://user-images.githubusercontent.com/6540262/186681791-49ba19ef-25e2-4353-bfbe-d3902bb986b1.png)

```
+-- monaco-editor@0.33.0
`-- monaco-languageclient@2.1.0
  `-- vscode@npm:@codingame/monaco-vscode-api@1.67.20
    `-- monaco-editor@0.33.0 deduped
```

I am using the monaco-editor/language-client in a vue application.

I added the `window.MonacoEnvironment` code below recently, so I'm trying to determine if this is the culprit.  Perhaps I'm doing something wrong here, but I was just trying to solve this warning:

![image](https://user-images.githubusercontent.com/6540262/186682482-fa3292a4-8dd0-42c7-8445-03ba79ee3127.png)

```ts
import editorWorker from "monaco-editor/esm/vs/editor/editor.worker?worker";
import jsonWorker from "monaco-editor/esm/vs/language/json/json.worker?worker";
import cssWorker from "monaco-editor/esm/vs/language/css/css.worker?worker";
import htmlWorker from "monaco-editor/esm/vs/language/html/html.worker?worker";
import tsWorker from "monaco-editor/esm/vs/language/typescript/ts.worker?worker";

export function useMonacoEnvironment() {
  // @ts-ignore
  if (!window.MonacoEnvironment) {
    console.debug("Setting up monaco environment");
    // @ts-ignore
    window.MonacoEnvironment = {
      getWorker(_: any, label: string) {
        if (label === "json") {
          return new jsonWorker();
        }
        if (label === "css" || label === "scss" || label === "less") {
          return new cssWorker();
        }
        if (label === "html" || label === "handlebars" || label === "razor") {
          return new htmlWorker();
        }
        if (label === "typescript" || label === "javascript") {
          return new tsWorker();
        }
        return new editorWorker();
      },
    };
  }
}
```

**Comment by CGNonofr** [comment]
The "missing service" error happens when there is multiple monaco editor versions involved

**Comment by colinblaise** [comment]
I am only seeing v 0.33.0 in my `yarn.lock` file.  But thanks I'll keep digging around with that in mind

**Comment by kaisalmen** [comment]
@CGNonofr FYI I have just created a couple of enhancement issues for creating framework related examples.

**Comment by CGNonofr** [comment]
> I am only seeing v 0.33.0 in my yarn.lock file. But thanks I'll keep digging around with that in mind

That's weird, because monaco-languageclient@3 has a dependency on monaco-vscode-api@1.69 which has a dependency on monaco-editor@0.34

**Comment by CGNonofr** [comment]
Oh you're not using the last version of monaco-languageclient! I don't know about your error, you should probably update though

**Comment by colinblaise** [comment]
Updated to v3 and still receiving the same error.  I'll figure this out.

```
+-- monaco-editor@0.34.0
`-- monaco-languageclient@3.0.1
  `-- vscode@npm:@codingame/monaco-vscode-api@1.69.9
    `-- monaco-editor@0.34.0 deduped
```

**Comment by CGNonofr** [comment]
I just realize it can happen if an editor is created before monaco-languageclient/monaco-vscode-api is loaded (because creating an editor initialize and freeze the services)

Otherwise I don't know, I'm interested it you find anything

**Comment by kaisalmen** [comment]
> Updated to v3 and still receiving the same error. I'll figure this out.

Plus, you don't need to define `monaco-editor` separately in your dependencies in the `package.json`? If so, remove it and only keep the dependency to `monaco-languageclient`.

**Comment by CGNonofr** [comment]
Since monaco-editor is imported directly, I think it's a good idea to have it in the package.json

**Comment by kaisalmen** [comment]
Yes, but you have to keep version in sync 🙂

**Comment by colinblaise** [comment]
I figured out how to reproduce my issue. 

I am instantiating multiple monaco editors on a page **without** setting up any language services. These are just simple read-only views.

![image](https://user-images.githubusercontent.com/6540262/186692232-f5645191-cd48-4b07-b027-8f22e45146e3.png)

If I then navigate to a page that instantiates an editor with language services, it throws the error.

It appears that there may be some initialization logic that only occurs once.  So if I load an editor without language services, and then load it with language services, the second load fails.

**Comment by colinblaise** [comment]
Oh, I didn't see your comment earlier about loading `monaco-languageclient/monaco-vscode-api`!

I'm not 100% sure what you mean by import `vscode`.

Simply add an import statement? `import vscode from "vscode"`? Or is `vscode` coming from somewhere in monaco-editor or monaco-languageclient

**Comment by CGNonofr** [comment]
Simply importing `vscode` should do the trick, importing `monaco-languageclient` should work as well

We use an alias from `vscode` to `@CodinGame/monaco-vscode-api`

**Comment by colinblaise** [comment]
That definitely fixed the issue.  Appreciate the help!

**Comment by CGNonofr** [comment]
@kaisalmen We should probably have a troubleshoot section somewhere, why do you think?

**Comment by kaisalmen** [comment]
> We should probably have a troubleshoot section somewhere, why do you think?

Good idea, new document linked from the README or we just add a section in the README itself

**Comment by CGNonofr** [comment]
Let' keep it simple as a section until it becomes too big

**Comment by mbui0327** [comment]
@wudstrand I try to do similar thing and wonder what python lsp server your are using here? Thanks.

**Comment by hoathaivan** [comment]
@CGNonofr 
I am trying to use @monaco-editor/react under nextjs to connect to python language server, and am having this problem.
![image](https://user-images.githubusercontent.com/7814259/190846339-21b7e775-e984-487a-96ca-1f0d573c50dd.png)
`import { useRef, useEffect } from "react";
import type { NextPage } from "next";
import styles from "../styles/Home.module.css";
import Editor, { useMonaco } from "@monaco-editor/react";
import normalizeUrl from "normalize-url";
import { toSocket, WebSocketMessageReader, WebSocketMessageWriter } from "vscode-ws-jsonrpc";
import {
    CloseAction,
    ErrorAction,
    MessageTransports,
    MonacoLanguageClient,
} from "monaco-languageclient";
import * as monaco from "monaco-editor";
type Monaco = typeof monaco;

const ReconnectingWebSocket = require("reconnecting-websocket");

const Home: NextPage = () => {
    // install Monaco language client services
    const LanguageId = "plaintext";
    const CODE = `
#ff0000 (red)
#00ff00 (green)
#0000ff (blue)
`;

    const onMount = (editor, monaco) => {
        console.log("onMount", editor, monaco);
        console.log("Mounted");

        function createUrl(hostname: string, port: number, path: string): string {
            // eslint-disable-next-line no-restricted-globals
            const protocol = location.protocol === "https:" ? "wss" : "ws";
            return normalizeUrl(`${protocol}://${hostname}:${port}${path}`);
        }

        function createWebsocket(socketUrl: string) {
            const socketOptions = {
                maxReconnectionDelay: 10000,
                minReconnectionDelay: 1000,
                reconnectionDelayGrowFactor: 1.3,
                connectionTimeout: 10000,
                maxRetries: Infinity,
                debug: false,
            };
            return new ReconnectingWebSocket.default(socketUrl, [], socketOptions);
        }

        function createLanguageClient(transports: MessageTransports): MonacoLanguageClient {
            return new MonacoLanguageClient({
                name: "Sample Language Client",
                clientOptions: {
                    // use a language id as a document selector
                    documentSelector: ["python"],
                    // disable the default error handler
                    errorHandler: {
                        error: () => ({ action: ErrorAction.Continue }),
                        closed: () => ({ action: CloseAction.DoNotRestart }),
                    },
                },
                // create a language client connection from the JSON RPC connection on demand
                connectionProvider: {
                    get: () => {
                        return Promise.resolve(transports);
                    },
                },
            });
        }

        // create the web socket
        const url = createUrl("localhost", 5000, "/");
        const webSocket = createWebsocket(url);

        webSocket.onopen = () => {
            console.log("Opening Web socket connection");
            const socket = toSocket(webSocket);
            const reader = new WebSocketMessageReader(socket);
            const writer = new WebSocketMessageWriter(socket);
            const languageClient = createLanguageClient({
                reader,
                writer,
            });

            languageClient
                .start()
                .then(() => console.log("language client started"))
                .finally(() => console.log("language client done"));

            reader.onClose(() => languageClient.stop());
        };
    };

    const beforeMount = (monaco) => {
        console.log("beforeMount", monaco);
        monaco.languages.register({
            id: "python",
            extensions: [".py"],
            aliases: ["PYTHON", "python", "py"],
        });
    };

    return (
        <div className={styles.container}>
            <Editor
                height="90vh"
                defaultLanguage={LanguageId}
                defaultValue={CODE}
                onMount={onMount}
                beforeMount={beforeMount}
            />
        </div>
    );
};

export default Home;
`

**Comment by CGNonofr** [comment]
Can you please create your own issue as it's not related?

**Comment by ChrisGilchrist** [comment]
I've ran into a similiar issue when on Angular 14 when migrating the libraries v 0.13.0 -> v4. 

We are currently seeing the following in the console:
![image](https://user-images.githubusercontent.com/24948056/196438480-288659ec-66fd-4e49-9aec-3a625aeadd2d.png)

During the migration, I have been using this code https://github.com/TypeFox/monaco-languageclient/blob/main/packages/examples/client/src/client.ts as a reference to base it off.

Any idea how to solve this? I've tried the above but not had much! :(

**Comment by kaisalmen** [comment]
@ChrisGilchrist did you check this already? https://github.com/TypeFox/monaco-languageclient/issues/412#issuecomment-1227303932

**Comment by ChrisGilchrist** [comment]
Yeah when running `npm list monaco-editor`, it shows the following:
![image](https://user-images.githubusercontent.com/24948056/196439896-d9e7a5cc-9e40-413e-a78e-dbe731aceaf3.png)

**Comment by kaisalmen** [comment]
You have `monaco-editor@0.34.1` that was just released (21h ago). Do you have another version in your list? Otherwise, please explicitly define `0.34.0` as dependency for now.
Do you import this 
```javascript
import * as monaco from "monaco-editor";
```
before anything else as also suggested here https://github.com/TypeFox/monaco-languageclient/issues/423#issuecomment-1255995975 ?

**Comment by kaisalmen** [comment]
@CGNonofr have you seen this:
https://github.com/microsoft/vscode/pull/163858/files#diff-0dfa7db579eface8250affb76bc88717725a121401d4d8598bc36b92b0b6ef62

That is the only change in 0.34.1. Does it affect `monaco-vscode-api`?

**Comment by CGNonofr** [comment]
I don't think so, does it introduce issues?

**Comment by kaisalmen** [comment]
Don't know, was just asking. Will you release a new version fixed to `0.34.1`? 

I will now set a direct dependency to `monaco-editor@0.34.1` for `monaco-languageclient` and upgrade that whenever needed. This way a transitive dependency should not change `monaco-editor` version it uses unless explicitly configured in a project. That reasoning ist correct, isn't?

**Comment by CGNonofr** [comment]
monaco-vscode-api synchronizes versions on vscode, not monaco-editor

I'm not sure yo understand what you mean, I see no reason to force a specific version of the editor

**Comment by kaisalmen** [comment]
> I'm not sure yo understand what you mean, I see no reason to force a specific version of the editor

ok, yes you are right. It must be defined on application/user level, not in the lib level (as it was done with 4.0.0)

For my understanding, why do you have direct dependency to `monaco-editor` defined here: https://github.com/CodinGame/monaco-vscode-api/blob/main/package.json#L177

**Comment by CGNonofr** [comment]
I admit it's not always clear to me when to use `dependencies` and `peerDependencies`, I'm not sure why you think that `the monaco-editor dependency here is not required.`

**Comment by kaisalmen** [comment]
> I'm not sure why you think that the monaco-editor dependency here is not required.

Because, the source code of the lib does not use it/import it (only the examples) and `monaco-vscode-api` declares it as dependency. Wouldn't it be best that `monaco-vscode-api` adds a peer dependency like (or any other fitting upper range):
```
"monaco": ">= 0.34.0 < 0.35.0"
```
If we have this peer-dependency in place, then users with non-matching `monaco-editor` versions will get warnings. WDYT? This could even be done here in addition, but it is better placed in `monaco-vscode-api` and only needed once IMO.

**Comment by CGNonofr** [comment]
> Because, the source code of the lib does not use it/import it

monaco-vscode-api? it does that A LOT

btw: isn't `>= 0.34.0 < 0.35.0` equivalent to `~0.34.0`?

**Comment by kaisalmen** [comment]
The version range was just an example, but being constraint here would be good, I guess. Image you jump with `monaco-vscode-api` to vs code 1.7x in the future, then you want to limited its compatibility to a specific range of `monaco-editor` as well, correct?

**Comment by ChrisGilchrist** [comment]
> You have `monaco-editor@0.34.1` that was just released (21h ago). Do you have another version in your list? Otherwise, please explicitly define `0.34.0` as dependency for now. Do you import this
> 
> ```js
> import * as monaco from "monaco-editor";
> ```
> 
> before anything else as also suggested here [#423 (comment)](https://github.com/TypeFox/monaco-languageclient/issues/423#issuecomment-1255995975) ?

I've tried following the above but still have not had much luck doing so. I unfortunately can't seem to figure out how to do the same with the library https://github.com/atularen/ngx-monaco-editor. Is it compatible with this library?

**Comment by kaisalmen** [comment]
I don't know. https://github.com/atularen/ngx-monaco-editor does not seem to be compatible with NG 14 as far as I can see.

Have you tried to build an new Angular component and basically add the init of [client example](https://github.com/TypeFox/monaco-languageclient/blob/main/packages/examples/client/src/client.ts#L29-72) to the init of the angular component? Add all imports as well and add the two functions from the example. Then all should be there.

We want to add basic [Angular](https://github.com/TypeFox/monaco-languageclient/issues/414) and [React](https://github.com/TypeFox/monaco-languageclient/issues/415) examples, but didn't get to it, yet.

**Comment by yuyic** [comment]
I ran repository example but got the error as following
![image](https://user-images.githubusercontent.com/785660/196636198-e512e6e7-1436-4f76-9ad4-0edced45504e.png)
I've tried to install monaco-editor@0.34.0 explicitly and downgrade to @codingame/monaco-vscode-api@1.69.10, but the error still appears
![image](https://user-images.githubusercontent.com/785660/196636470-382e2a31-8884-430f-a17f-41f876e1e8ed.png)

**Comment by CGNonofr** [comment]
This error can occur if you instantiate an editor BEFORE loading monaco-languageclient or monaco-vscode-api

**Comment by yuyic** [comment]
> This error can occur if you instantiate an editor BEFORE loading monaco-languageclient or monaco-vscode-api

The error occurs while creating new MonacoLanguageClient
![image](https://user-images.githubusercontent.com/785660/196650560-5444caa4-99c5-44ea-ac1c-7d2d6caeb68f.png)

**Comment by CGNonofr** [comment]
Yes, was an editor created before?

**Comment by yuyic** [comment]
> Yes, was an editor created before?

No I've commented out `monaco.editor.create...`, I ran the example by `npm run dev`(vite)

**Comment by CGNonofr** [comment]
Oh I didn't realize you were talking about the example

I just checked and everything seems to run fine on main HEAD

**Comment by kaisalmen** [comment]
I just added a [Volta section](https://github.com/TypeFox/monaco-languageclient#volta) in the new Troubleshooting in the README. When using Volta mismatching node/npm versions are no longer an issue 😎

**Comment by kaisalmen** [comment]
I close this because #415 was implemented via #436. We have an official react-client example now.
